configure.ac

```diff
@@ -3679,7 +3685,13 @@ int main(void)
 }]])],
 [ac_cv_aligned_required=no],
 [ac_cv_aligned_required=yes],
-[ac_cv_aligned_required=yes])
+[
+# "yes" changes the hash function to FNV, which breaks Numba.
+if test "$ac_sys_system" = "Linux-android" || test "$ac_sys_system" = "iOS"; then
+  ac_cv_aligned_required=no
+else
+  ac_cv_aligned_required=yes
+fi])
```

Comment:
I'm not 100% sure about this change (for the iOS case, anyway). 

Configure currently autodetects this on iOS as "ac_cv_aligned_required=yes", and *doesn't* define `Py_HASH_ALGORITHM`. That configuration (along with the other changes that haven't been submitted yet) passes the full CPython test suite. 

Introducing this change broke `test_posix.test_confstr`; but the same test passes when re-running the same test in isolation. I'm not 100% confident if this is an indicator of a problem with this setting, or a transient problem, but it's at least cause for concern.

Regardless, my understanding is that JIT compilation on iOS isn't possible (without jailbreaking or similar), as iOS doesn't allow executable pages to be modified, so running Numba may not be possible on iOS.

I've also seen [references online](https://stackoverflow.com/a/29708802) that suggest that while unaligned access is generally possible on ARM, it's slower, because unaligned writes can't be atomic. I don't have a more official source for this, though.

On that basis, my inclination is to *not* include iOS as part of this check.

Reply:
> Introducing this change broke `test_posix.test_confstr`; but the same test passes when re-running the same test in isolation.

Interesting: it looks like `confstr` doesn't exist on Android, so I've never run that test. What was the failure message? And are you sure you didn't make any other changes at the same time?

Reply:
The failure message was:
```
test test_posix failed -- Traceback (most recent call last):
  File "/Users/rkm/Library/Developer/XCTestDevices/7258455F-A398-4EEE-B736-B1F98D276CDE/data/Containers/Bundle/Application/355D8462-61DD-4B65-B23E-641FF7128201/iOSTestbed.app/python/lib/python3.13/test/test_posix.py", line 569, in test_confstr
    self.assertEqual(len(posix.confstr("CS_PATH")) > 0, True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: False != True
```

I'm fairly certain there were no additional changes, other than the stash of patches I need to get the tests passing on iOS at all. I applied this one part of your patch, reconfigured, ran the tests, got the failure; ran the test in isolation, and it passed; removed the patch, reconfigured, ran the tests, and it passed.

I agree it's a weird failure though. I can't think of any obvious reason why such a simple test would fail - my only guess was that something might have gone awry as a result of a copy not being aligned at in the middle of a large test run where an executable is doing lots of copies... but then, I would have expected to see a lot of *other* failures as well.

Reply:
OK, I've removed iOS from this check.